src/Organisms/DogData/DogData.js

```diff
@@ -1,55 +1,53 @@
-import { useEffect, useState } from 'react';
+import PropTypes from 'prop-types';
+import { useNavigate } from 'react-router-dom';
 
 import ColumnWrapper from '../../Templates/ColumnWrapper/ColumnWrapper';
-import DOG_DATA_TEMPLATE from '../../Consts/DogDataCONSTS';
+import doggos from '../../Data/MongoDBMock/doggos';
+import renderDogData from '../../Tools/renderDogData';
 import DataLine from '../../Atoms/DataLine/DataLine';
-import PropTypes from 'prop-types';
 import SpecialButton from '../../Atoms/SpecialButton/SpecialButton';
 import SpecialButtonsContainerStyled from '../../Molecules/SpecialButtonsContainer/SpecialButtonsContainerStyled';
-import doggos from '../../Data/MongoDBMock/doggos';
-import { getDataFormatDdMonthYyy } from '../../Tools/TimeFunctions';
-import { useNavigate } from 'react-router-dom';
+import { useContext, useEffect, useState } from 'react';
+import { UserDataContext } from '../../Context/UserDataContext';
+import { DogContext } from '../../Context/DogContext';
 
 const DogData = ({ id }) => {
   let navigate = useNavigate();
   const [dogData, setDogData] = useState(null);
   const [isPending, setIsPending] = useState(true);
+  const { state } = useContext(UserDataContext);
+  const { dogState } = useContext(DogContext);
 
   useEffect(() => {
-    // @TODO fetching data from database in the future here
     setDogData(doggos.find((dog) => dog.dogId === id));
     setIsPending(false);
   }, []);
 
-  const dogDataRender = {};
-
-  if (dogData) {
-    Object.keys(DOG_DATA_TEMPLATE).forEach(
-      (property) =>
-        (dogDataRender[DOG_DATA_TEMPLATE[property]] =
-          dogData[property] || 'brak danych'),
-    );
-    dogDataRender['Data urodzenia'] = `${getDataFormatDdMonthYyy(
-      dogDataRender['Data urodzenia'],
-    )}`;
-  }
-
   const handleEdit = (event) => {
     event.preventDefault();
-    navigate(`/add-dog-form`);
-    //navigate musi przekazać dane psa (id) do formularza który wypełni sobie inputy value z bazy
-    // danych
+    navigate(`/add-dog-form/`, {
+      state: {
+        text: 'Formularz - edycja',
+        label: `${dogData.dogName}`,
+      },
+    });
   };
+
   const handleConfirm = (event) => {
     event.preventDefault();
-    const idOfOwnerFromContext = '@TODO in the future';
-    navigate(`/participant-data/${idOfOwnerFromContext}`);
-    //navigate musi przekazać id właściela psa, żeby pokazać nam kolejny ekran z wypełnionymi
-    // danymi właściciela
+    !dogState.chosenDog &&
```

Comment:
![image](https://user-images.githubusercontent.com/92270179/153217352-66925ef6-b065-46e4-a318-a2e19328790d.png)
dog will be chosen at both flow, it should be changed to chosenContest / selectedContest.
But it that PR there is a lot of other changes, for me we can merge it, I'll create proper elements in Context and then you'll fix it